great_expectations/checkpoint/checkpoint.py

```diff
@@ -152,6 +157,12 @@ def run(
 
         run_id = run_id or RunIdentifier(run_name=run_name, run_time=run_time)
 
+        # Ensure that validations dicts have the most specific id available
+        # (default to Checkpoint's default_validation_id if no validations were passed in the signature)
+        if using_default_validation:
+            for validation in validations:
+                validation["id_"] = self.config.default_validation_id
```

Comment:
This object may be a dict or a `CheckpointValidaitonConfig` - using `[]` notation allows us to work with both.